### DISCUSSION TITLE
Error: Hydration failed because the initial UI does not match what was rendered on the server.

### DISCUSSION THREAD

**Discussion by mihaimazareanu** [neuefische/questions#341] [Web] [answered]
Hi all,

I keep getting the above error when I start the app or whenever I reload the page.

The whole text of the error message is this:

    Unhandled Runtime Error
    Error: Hydration failed because the initial UI does not match what was rendered on the server.

    See more info here: https://nextjs.org/docs/messages/react-hydration-error

    Call Stack
    throwOnHydrationMismatch
    node_modules/react-dom/cjs/react-dom.development.js (12507:0)
    tryToClaimNextHydratableInstance
    node_modules/react-dom/cjs/react-dom.development.js (12535:0)
    updateHostComponent
    node_modules/react-dom/cjs/react-dom.development.js (19902:0)
    beginWork
    node_modules/react-dom/cjs/react-dom.development.js (21618:0) 
    HTMLUnknownElement.callCallback
    node_modules/react-dom/cjs/react-dom.development.js (4164:0)
    Object.invokeGuardedCallbackDev
    node_modules/react-dom/cjs/react-dom.development.js (4213:0)
    invokeGuardedCallback
    node_modules/react-dom/cjs/react-dom.development.js (4277:0)
    beginWork$1
    node_modules/react-dom/cjs/react-dom.development.js (27451:0)
    performUnitOfWork
    node_modules/react-dom/cjs/react-dom.development.js (26557:0)
    workLoopSync
    node_modules/react-dom/cjs/react-dom.development.js (26466:0)
    renderRootSync
    node_modules/react-dom/cjs/react-dom.development.js (26434:0)
    performConcurrentWorkOnRoot
    node_modules/react-dom/cjs/react-dom.development.js (25738:0)
    workLoop
    node_modules/scheduler/cjs/scheduler.development.js (266:0)
    flushWork
    node_modules/scheduler/cjs/scheduler.development.js (239:0)
    MessagePort.performWorkUntilDeadline
    node_modules/scheduler/cjs/scheduler.development.js (533:0)

I have been researching this error message since Monday and I am not any smarter than I was then...

I understand that the issue is caused by a difference between what the client browser renders and what is on the server, but I have no idea how to solve the issue. I have tried using useEffect, but I stumbled into other problems (probably I didn't use useEffect correctly? ). 

I don't have any api routes yet, as I didn't start working on the databank yet.

Anyone know how to solve this issue? It's driving me crazy... :(

**Comment by ahohnsen** [comment]
Please push the current version of your app and post a link to the PR. :)

**Reply by mihaimazareanu** [reply]
Ok, since you obviously can't read my mind 😛  here is the PR. 😄 

https://github.com/mihaimazareanu/capstone-project-kidsfi/pull/10

**Comment by actyralf** [comment, marked as answer]
Hi Mihai, you must NOT use the `<Body>` component in your pages or components, this will lead to two `<body>` tags in the rendered HTML which is invalid. If you remove `<Body>`from all occurrences EXCEPT `_document.js`, everything will work fine.

**Reply by mihaimazareanu** [reply]
Thank you Ralf, it is so nice to not have this error anymore. 😄
